test/ng/directive/inputSpec.js

```diff
@@ -1966,28 +1974,32 @@ describe('input', function() {
       dealoc(formElm);
     });
 
-    it('should not reuse the hour part of a previous date object after changing the timezone', function() {
+    it('should not reuse the hours part of a previous date object after changing the timezone', function() {
       var inputElm = helper.compileInput('<input type="date" ng-model="value" ng-model-options="{timezone: \'UTC\'}" />');
 
       helper.changeInputValueTo('2000-01-01');
+      // The Date parser sets the hours part of the Date to 0 (00:00)
       expect(+$rootScope.value).toBe(Date.UTC(2000, 0, 1, 0));
 
       // Change the timezone offset so that the display date is a day earlier
       // This does not change the model, but our implementation
       // internally caches a Date object with this offset
-      // and re-uses it if part of the date changes
+      // and re-uses it if part of the Date changes.
+      // See https://github.com/angular/angular.js/commit/1a1ef62903c8fdf4ceb81277d966a8eff67f0a96
       inputElm.controller('ngModel').$overrideModelOptions({timezone: '-0500'});
       $rootScope.$apply(function() {
         $rootScope.value = new Date(Date.UTC(2000, 0, 1, 0));
       });
       expect(inputElm.val()).toBe('1999-12-31');
-
-      // Emptying the input should clear the cached date object
-      helper.changeInputValueTo('');
+      // At this point, the cached Date has its hours set to to 20 (00:00 - 05:00 = 19:00)
```

Comment:
I am confused 😕
Are you saying that 19 === 20?

Reply:
No, just a result of trying to understand timezones. If you log a date, you always get it with the local timezone offset which is +1 for me, which means 20. But 19 is correct.